challenge_9/java/jdfurlan/src/Squares.java

```diff
@@ -0,0 +1,50 @@
+
+import java.util.ArrayList;
+
+public class Squares {
+
+	public static void main(String[] args) {
+		int[] arr = {-9,4,9};
+
+		ArrayList<Integer> myList = orderSquares(arr);
+		for (int i : myList) {
+			System.out.print(i + " ");
+		}
+
+	}
+
+	private static ArrayList orderSquares(int[] arr) {
+		ArrayList<Integer> result = new ArrayList<Integer>();
+		ArrayList<Integer> negatives = new ArrayList<Integer>();
```

Comment:
Utilize java's LinkedList for the negatives and positives. That way you can use them like a queue. Specifically, adding to the 0 index of the negatives array will cause the array to be recopied every time to shift all the contained elements. A linked list on the other hand would just add a new node to the front with little overhead.